eclipse-theia/theia
# Add support for conditional exception breakpoints

#### What it does

Closes #12444 by adding support for conditional exception breakpoints to Theia:

<img width="331" src="https://user-images.githubusercontent.com/8298491/233688761-05e4ad14-7b46-42aa-ae46-aacf177ffdaf.png">

#### How to test

Try to debug a simple Node.js program like this:

```js
class MyError extends Error {
    constructor(message) {
        super(message);
        this.name = this.constructor.name;
    }
}

try {
    throw new Error();
} catch(e) {
    console.error(e);
}
try {
    throw new MyError();
} catch(e) {
    console.error(e);
}
throw new Error();
```

Enable both `Caught Exceptions` and `Uncaught Exceptions` in `Breakpoints` view. Try setting a condition like `error.name == "MyError"` using the context menu item `Edit Condition...` for either of the exception breakpoints. Verify that exception filtering works as expected while debugging. Also, verify that exception breakpoints' condition settings are properly restored upon workbench restart. Try removing the condition by editing it again and blanking out the input field.

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
